src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceService.kt

```diff
@@ -74,6 +78,21 @@ class LicenceService(
     licenceRepository.saveAndFlush(updatedLicence)
   }
 
+  @Transactional
+  fun updateBespokeConditions(licenceId: Long, request: BespokeConditionRequest) {
+    licenceRepository
+      .findById(licenceId)
+      .orElseThrow { EntityNotFoundException("$licenceId") }
+
+    bespokeConditionRepository.deleteByLicenceId(licenceId)
+
+    request.conditions.forEachIndexed { index, condition ->
+      bespokeConditionRepository.saveAndFlush(
```

Comment:
Instead of saving and flushing all entities sequentially, you could build up a List of entities, and call `saveAllAndFlush(conditionsList)`

Reply:
Yes, good call. Would you mind if I do that when I'm next in the API? I'm keen to get this in and crack on with the next bits in the UI, which needs the type imports from here.

Reply:
Sure thing